src/shipit_code_coverage/shipit_code_coverage/codecov.py

```diff
@@ -207,23 +214,28 @@ def get_commit():
             raise Exception('Mercurial commit is not available yet on mozilla/gecko-dev.')
         return ret
 
-    def rewrite_jsvm_lcov(self, zip_file_path):
-        with self.build_finished_cv:
-            while not self.build_finished:
-                self.build_finished_cv.wait()
+    def rewrite_jsvm_lcov(self, platform_name, zip_file_path):
+        with self.configure_finished_cv:
+            while not self.configure_finished:
+                self.configure_finished_cv.wait()
 
         out_dir = zip_file_path[:-4]
         out_file = out_dir + '.info'
 
         with ZipFile(zip_file_path, 'r') as z:
             z.extractall(out_dir)
-
-        run_check([
-            'gecko-env', './mach', 'python',
-            'python/mozbuild/mozbuild/codecoverage/lcov_rewriter.py',
-            os.path.abspath(out_dir),
-            '--output-file', os.path.abspath(out_file)
-        ], cwd=self.repo_dir)
+            files = z.namelist()
+            if len(files) == 1 and files[0] == 'jsvm_lcov_output.info':
+                # The file was rewritten on the test machine.
```

Comment:
This is only needed until all test machines have moved to rewriting LCOV files. When that happens, we will be able to completely remove the LCOV rewriting code (https://github.com/mozilla-releng/services/issues/793).